loadgen/bindings/python_api.cc

```diff
@@ -0,0 +1,182 @@
+    // TODO: Get rid of copies.
+    load_samples_to_ram_cb_(
+          std::vector<QuerySample>(samples, samples+sample_count));
+  }
+  void UnloadSamplesFromRam(QuerySample* samples,
+                            size_t sample_count) override {
+    pybind11::gil_scoped_acquire gil_acquirer;
+    // TODO: Get rid of copies.
+    unload_samlpes_from_ram_cb_(
+          std::vector<QuerySample>(samples, samples+sample_count));
+  }
+
+  // TODO(brianderson): Accuracy Metric API.
+  void ResetAccuracyMetric() override {}
+  void UpdateAccuracyMetric(uint64_t sample_index, void* response_data,
+                            size_t response_size) override {}
+  double GetAccuracyMetric() override {return 0;}
+  std::string HumanReadableAccuracyMetric(double metric_value) override {
+    return "TODO: AccuracyMetric";
+  }
+
+ private:
+  std::string name_;
+  size_t total_sample_count_;
+  size_t performance_sample_count_;
+  LoadSamplesToRamCallback load_samples_to_ram_cb_;
+  UnloadSamplesFromRamCallback unload_samlpes_from_ram_cb_;
+};
+
+}  // namespace
+
+namespace py {
+  void* ConstructSUT(std::string name, IssueQueryCallback issue_cb) {
+    SystemUnderTestTrampoline* sut = new SystemUnderTestTrampoline(name, issue_cb);
+    return reinterpret_cast<void*>(sut);
+  }
+
+  void DestroySUT(void* sut) {
+    SystemUnderTestTrampoline* sut_cast =
+        reinterpret_cast<SystemUnderTestTrampoline*>(sut);
+    delete sut_cast;
+  }
+
+  void* ConstructQSL(std::string name,
+                     size_t total_sample_count, size_t performance_sample_count,
+                     LoadSamplesToRamCallback load_samples_to_ram_cb,
+                     UnloadSamplesFromRamCallback unload_samlpes_from_ram_cb) {
+    QuerySampleLibraryTrampoline* qsl = new QuerySampleLibraryTrampoline(
+        name, total_sample_count, performance_sample_count,
+        load_samples_to_ram_cb, unload_samlpes_from_ram_cb);
+    return reinterpret_cast<void*>(qsl);
+  }
+
+  void DestroyQSL(void* qsl) {
+    QuerySampleLibraryTrampoline* qsl_cast =
+        reinterpret_cast<QuerySampleLibraryTrampoline*>(qsl);
+    delete qsl_cast;
+  }
+
+  // Parses commandline.
+  void StartTest(void* sut, void* qsl, std::string command_line) {
+    pybind11::gil_scoped_release gil_releaser;
+    SystemUnderTestTrampoline* sut_cast =
+        reinterpret_cast<SystemUnderTestTrampoline*>(sut);
+    QuerySampleLibraryTrampoline* qsl_cast =
+        reinterpret_cast<QuerySampleLibraryTrampoline*>(qsl);
+    mlperf::TestSettings default_settings;
+    mlperf::StartTest(sut_cast, qsl_cast, default_settings);
+  }
+
+  // TODO: Get rid of copies.
+  void QueryComplete(QueryId query_id,
+                     std::vector<QuerySampleResponse> responses) {
+    pybind11::gil_scoped_release gil_releaser;
+    mlperf::QueryComplete(query_id, responses.data(), responses.size());
+  }
+}  // namespace py
+}  // namespace mlperf
+
+PYBIND11_MODULE(mlpi_loadgen, m) {
```

Comment:
mlpi_loadgen is unclear, how about mlperf_loadgen ?

Reply:
Done.